File: 733-flood-fill/flood-fill.py
class Solution(object):
    def floodFill(self, image, sr, sc, color):
        """
        :type image: List[List[int]]
        :type sr: int
        :type sc: int
        :type color: int
        :rtype: List[List[int]]
        """
        m=len(image)
        n=len(image[0])
        intial=image[sr][sc]
        if intial == color:
            return image
        def dfs(i,j):
            if i>=m or i<0 or j>=n or j<0 or image[i][j]!=intial:
                return
            image[i][j]=color
            dfs(i+1,j)
            dfs(i-1,j)
            dfs(i,j+1)
            dfs(i,j-1)
        dfs(sr,sc)
        return image
        

File: 733-flood-fill/flood-fill.cpp
class Solution {
public:
    void dfs(vector<vector<int>>& grid,int i,int j,int color,int org){

        if(i<0 || j<0 || i>=grid.size() || j>=grid[0].size() || grid[i][j]!=org) return;

        grid[i][j]=color;
        dfs(grid,i+1,j,color,org);
        dfs(grid,i-1,j,color,org);
        dfs(grid,i,j+1,color,org);
        dfs(grid,i,j-1,color,org);
    }
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int color) {
        int org=image[sr][sc];
        if(color==org) return image;
        dfs(image,sr,sc,color,org);
        return image;
    }
};